plasmapy/simulation/tests/test_particle_tracker.py

```diff
@@ -0,0 +1,53 @@
+"""
+Tests for particle_tracker.py
+"""
+import astropy.units as u
+import numpy as np
+
+from plasmapy.formulary.lengths import gyroradius
+from plasmapy.particles import CustomParticle
+from plasmapy.plasma.grids import CartesianGrid
+from plasmapy.simulation.particle_tracker import (
+    MemoryIntervalSaveRoutine,
+    ParticleTracker,
+    TimeElapsedStopCondition,
+)
+
+
+class TestParticleTracker:
+    def test_gyroradius(self):
+        """Test to ensure particles maintain their gyroradius over time"""
+        L = 1 * u.km
+        num = 2
+
+        grid = CartesianGrid(-L, L, num=num)
+
+        grid_shape = (num,) * 3
+        B_strength = 1 * u.T
+        Bz = np.full(grid_shape, B_strength) * u.T
+
+        grid.add_quantities(B_z=Bz)
+
+        simulation = ParticleTracker(
+            grid,
+            req_quantities=["E_x", "E_y", "E_z", "B_x", "B_y", "B_z"],
+            verbose=False,
+        )
+
+        x = np.array([[1, 0, 0], [2, 0, 0], [3, 0, 0]]) * u.m
```

Comment:
Great test! To avoid hardcoding these numbers, you could define v as v_x, determine the corresponding gyroradius r_L using the plasmapy function (for this particle + bfield + perp velocity) R_L, then set the initial position as (0, R_L, 0). That way the particle should orbit around the origin without hardcoding these values.